pumpkin/src/server/connection_cache.rs

```diff
@@ -42,28 +42,26 @@ pub struct CachedStatus {
     // Keep in mind that we must parse this again, when the StatusResponse changes which usually happen when a player joins or leaves
     status_response_json: String,
 }
+/// Cached Server brand buffer so we don't have to rebuild them every time a player joins
 
-pub struct CachedBranding {
-    /// Cached Server brand buffer so we don't have to rebuild them every time a player joins
-    cached_server_brand: Vec<u8>,
-}
+pub struct CachedBranding(());
 
 impl CachedBranding {
     pub fn new() -> Self {
-        let cached_server_brand = Self::build_brand();
-        Self {
-            cached_server_brand,
-        }
+        Self(())
     }
+
+    #[allow(clippy::unused_self)]
     pub fn get_branding(&self) -> CPluginMessage {
-        CPluginMessage::new("minecraft:brand", &self.cached_server_brand)
-    }
-    fn build_brand() -> Vec<u8> {
-        let brand = "Pumpkin";
-        let mut buf = vec![];
-        let _ = VarInt(brand.len() as i32).encode(&mut buf);
-        buf.extend_from_slice(brand.as_bytes());
-        buf
+        static BRANDING: LazyLock<Box<[u8]>> = LazyLock::new(|| {
+            let brand = "Pumpkin";
+            let mut buf = vec![];
+            let _ = VarInt::try_from(brand.len()).unwrap().write_to(&mut buf);
+            buf.extend_from_slice(brand.as_bytes());
+            buf.into_boxed_slice()
+        });
+
```

Comment:
I am not ok with this change, and I think clippy pointed you towards part of why I'm not ok with the change.

For one thing, this forces all `CachedBranding` to be the same, and only initialized once. All branding is contained within one singleton, global brand which isn't obvious from the `CachedBranding`'s API. If I were to see a `CachedBranding` being used elsewhere, it just wouldn't be obvious that I can construct one literally anywhere I'd like and it's the same everywhere.

For another reason, you've limited the scope of `BRANDING` to `get_branding()`, this means that it's completely impossible to modify the branding at runtime, despite it being lazily evaluated. There is nothing that can modify or update the branding later, and that I think is pretty pointless. If we're going to force it to be unmodifiable at runtime, we might as well make the whole expression `const` and compile time no?

If this static, singleton version of branding were to be done (which I'm not principally against) I would rather there be a `static BRANDING: LazyLock<CachedBranding> =...` lazy static rather than it be an internal implementation detail. 

I'm not sure why clippy didn't warn you, but if you have an empty struct, just use `struct CachedBranding;`

tl;dr while the actual implementation of branding isn't 100% right now, this is solidly a step backwards towards either making branding compile time or towards making it modifiable at runtime.

Reply:
> If we're going to force it to be unmodifiable at runtime, we might as well make the whole expression const and compile time no?

we could, I just used the path of lowest resistance

> I am not ok with this change, and I think clippy pointed you towards part of why I'm not ok with the change.
oh yeah for sure, I was going to talk about how a lot of api's seem hacked together 

my idea isn't to make lots of drastic changes, thats why I didn't modify the api, and I think this should have been just a Branding struct, not a `CachedBranding` that is locked to just the value of pumpkin

Reply:
If you'd like to update this API that would be a separate PR IMHO. I don't like the interior statics and how the branding is now completely locked to a single function on a tuple struct. I would prefer it to not be static at all, but if it is static, this is not exactly flexible.